test/query.test.js

```diff
@@ -2708,17 +2710,86 @@ describe('query', () => {
     }
   })
 
-  // TODO Create tests once work is done in Core
-  test.skip('current_identity', () => {})
+  // API v4 AccessProvider functions
+  // TODO: Add tests for JWTs using Auth0/Okta
+
+  test('current_identity returns object when authed', async () => {
+    // Create a new Collection
+    const newCollection = await client.query(
+      query.Create(collectionRef, { credentials: { password: 'sekrit' } })
+    )
+
+    // Login with permissions for newly created Collection
+    const newLogin = await client.query(
+      query.Login(newCollection['ref'], { password: 'sekrit' })
+    )
+
+    // Get the newly scoped client
+    const newClient = await util.getClient({ secret: newLogin['secret'] })
+
+    // Retrieve the current identity
+    const currentIdentity = await newClient.query(query.CurrentIdentity())
+    expect(currentIdentity).toEqual(newCollection.ref)
+    expect(currentIdentity).toEqual(newLogin.instance)
+    expect(currentIdentity).toBeInstanceOf(Object)
+  })
+
+  test('current_identity fails when no identity present', async () => {
+    try {
+      await adminClient.query(query.CurrentIdentity())
+    } catch (err) {
+      expect(err).toBeInstanceOf(errors.BadRequest)
+    }
+  })
 
-  // TODO Add test once Core work has been done
-  test.skip('has_current_identity', () => {})
+  test('has_current_identity returns true when identity present', async () => {
+    // Create a new Collection
+    const newCollection = await client.query(
+      query.Create(collectionRef, { credentials: { password: 'sekrit' } })
+    )
 
-  // TODO Finish test after Core work is done
-  test.skip('current_token', () => {})
+    // Login with permissions for newly created Collection
+    const newLogin = await client.query(
+      query.Login(newCollection['ref'], { password: 'sekrit' })
+    )
+
+    // Get the newly scoped client
+    const newClient = await util.getClient({ secret: newLogin['secret'] })
+
+    // Retrieve the current identity
+    const currentIdentity = await newClient.query(query.CurrentIdentity())
+    expect(currentIdentity).toBeTruthy()
+  })
+
+  test('has_current_identity returns false without identity present', async () => {
+    const currentIdentity = await adminClient.query(query.HasCurrentIdentity())
+    expect(currentIdentity).toBeFalsy()
+  })
+
+  test('current_token returns proper token', async () => {
+    const currentToken = await adminClient.query(query.CurrentToken())
+    expect(currentToken).toEqual(adminKey.ref)
+  })
+
+  test('current_token fails when unauthenticated', async () => {
+    try {
+      const currentToken = await client.query(query.CurrentToken())
```

Comment:
using `client` did not change the behavior, `CurrentToken()` is just returning the key ref, regardless the key type you are using, if you don't call something that will actually force the test to fail after the function call, you will never know that the function is actually working, not failing

Reply:
I am checking that `CurrentToken` works in a separate test above: https://github.com/fauna/faunadb-js/pull/340/files#diff-c2e271250d0fe43a2f2478970f73543eR2769

The `client.query` call itself is causing a failure, which is why the assertion in the `catch` is passing. This is the response I get from Core when logging out `err`:

```
      BadRequest {
        name: 'BadRequest',
        message: 'invalid token',
        description: 'Token metadata is not accessible.',
        requestResult:
         RequestResult {
           method: 'POST',
           path: '',
           query: null,
           requestRaw: '{"current_token":null}',
           requestContent: Expr { raw: [Object] },
           responseRaw:
            '{"errors":[{"position":[],"code":"invalid token","description":"Token metadata is not accessible."}]}',
           responseContent: { errors: [Array] },
           statusCode: 400,
           responseHeaders:
            { connection: 'keep-alive',
              'content-length': '101',
              'content-type': 'application/json;charset=utf-8',
              'x-compute-ops': '1',
              'x-faunadb-build': '20.09.00.nightly20200917-81798af',
              'x-query-bytes-in': '22',
              'x-query-bytes-out': '101',
              'x-query-time': '4',
              'x-read-ops': '0',
              'x-storage-bytes-read': '0',
              'x-storage-bytes-write': '0',
              'x-txn-retries': '0',
              'x-txn-time': '1600434387785000',
              'x-write-ops': '0' },
           startTime: 1600434387779,
           endTime: 1600434387786 } }
```

Reply:
I see why this is returning the error, but this is not related to the fact it's unauthenticated. It's because the `client` is using a key that points to a sub database, however, the key lives in the parent database (because it was created with the [rootClient](https://github.com/fauna/faunadb-js/blob/d44f3c956eb2c1e3efc802e9d55c65cd65b87e22/test/util.js#L129-L144)), so that key is not accessible from the db it's pointing.